Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-problems.tsx

```diff
@@ -52,15 +66,36 @@ const ProblemsProvider = ({ children }: IProblemsProviderProps) => {
         saveAttachment,
     } = useHttp(getDownloadProblemResourceUrl as UrlType, { id: problemResourceIdToDownload });
 
+    const normalizeProblems = useMemo(
+        () => normalizeOrderBy(problems),
+        [ problems ],
+    );
+
     const selectProblemById = useCallback(
         (problemId: number) => {
-            const newProblem = problems.find((p) => p.id === problemId);
+            const newProblem = normalizeProblems.find((p) => p.id === problemId);
 
             if (newProblem) {
                 setCurrentProblem(newProblem);
+                const { orderBy } = newProblem;
+
+                changeHash(orderBy);
             }
         },
-        [ problems ],
+        [ changeHash, normalizeProblems ],
+    );
+
+    const problemFromHash = useMemo(
+        () => {
+            const hashIndex = Number(params) - 1;
+            return normalizeProblems[hashIndex];
+        },
+        [ normalizeProblems, params ],
+    );
+
+    const loadFromHash = useMemo(
```

Comment:
`loadFromHash` is not good enough naming. It implies "Load this problem from the hash", when, actually, it returns "Is the problem loaded from hash?"

`isLoadedFromHash` is better naming.